sdk/src/settings.rs

```diff
@@ -218,40 +233,327 @@ impl Default for Verify {
+}
+
 // Settings for Builder API options
 #[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
-#[cfg_attr(feature = "json_schema", derive(JsonSchema))]
 #[allow(unused)]
 pub(crate) struct Builder {
-    auto_thumbnail: bool,
+    /// Claim generator info that is automatically added to the builder.
+    ///
+    /// Note that this information will prepend any claim generator info
+    /// provided explicitly to the builder.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    claim_generator_info: Option<Vec<ClaimGeneratorInfo>>,
+
+    /// Various settings for configuring automatic thumbnail generation.
+    thumbnail: ThumbnailSettings,
+    //
+    // Whether to automatically generate a c2pa.created assertion when
+    // TODO: describe when
+    // auto_created_action: bool,
+
+    // Whether to automatically generate a c2pa.opened assertion when
+    // the parent ingredient isn't supplied.
+    // auto_opened_action: bool,
+
+    // Whether to automatically generate a c2pa.placed assertion when
+    // ingredients with the componentOf relationship are added.
+    // auto_placed_action: bool,
+    // TODO:
+    // /// Indicates if the actions assertion
+    // ///
+    // /// https://c2pa.org/specifications/specifications/2.2/specs/C2PA_Specification.html#_all_actions_included
+    // all_actions_included: bool,
+    // // actions
+    // // action_templates
+    // // vs.
+    // // inherit_assertions
+    // // add_assertions
 }
 
 impl Default for Builder {
     fn default() -> Self {
         Self {
-            auto_thumbnail: true,
+            // TODO: add c2pa-rs?
+            claim_generator_info: None,
+            thumbnail: Default::default(),
         }
     }
 }
 
-impl SettingsValidate for Builder {}
+impl SettingsValidate for Builder {
+    fn validate(&self) -> Result<()> {
+        self.thumbnail.validate()
+    }
+}
 
-// Settings configuration for C2PA-RS.  Default configuration values
-// are lazy loaded on first use.  Values can also be loaded from a configuration
-// file or by setting specific value via code.  There is a single configuration
-// setting for the entire C2PA-RS instance.
-#[derive(Clone, Debug, Default, Deserialize, PartialEq, Serialize)]
-#[cfg_attr(feature = "json_schema", derive(JsonSchema))]
+// TODO: provide example config
+/// Settings for configuring all aspects of c2pa-rs.
+///
+/// [Settings::default] will be set thread-locally by default. To override these default fields,
+/// call [Settings::set_thread_local] with a new [Settings]. To obtain the thread local [Settings]
+/// call [Settings::thread_local].
+#[derive(Clone, Debug, Default, Deserialize, Serialize)]
 #[allow(unused)]
 pub struct Settings {
     trust: Trust,
     core: Core,
     verify: Verify,
     builder: Builder,
+
+    #[serde(skip)]
+    config: Config,
 }
 
 impl Settings {
-    #[allow(unused)]
+    /// Returns the current [Settings] for the local thread.
+    pub fn thread_local() -> Option<Settings> {
```

Comment:
How is this different than get_settings?
I would not call this thread_local because it means nothing.  The thread has no way to know of any other settings so IT IS the settings.  And technically it is a snapshot of the settings as such I am inclined to do away this this function since it could lead to some tricky bugs.  Just let user request the info they need from the Settings structure withouth holding a copy that could be stale.